crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -302,38 +296,46 @@ use std::{
 ///
 /// # bevy_ecs::system::assert_is_system(my_system);
 /// ```
-pub trait WorldQuery {
-    type Fetch: for<'world, 'state> Fetch<'world, 'state, State = Self::State>;
+pub trait WorldQuery: for<'w> WorldQueryGats<'w, _State = Self::State> {
     type State: FetchState;
-    type ReadOnlyFetch: for<'world, 'state> Fetch<'world, 'state, State = Self::State>
-        + ReadOnlyFetch;
+
+    fn shrink<'wlong: 'wshort, 'wshort>(item: QueryItem<'wlong, Self>) -> QueryItem<'wshort, Self>;
 }
 
-pub type QueryItem<'w, 's, Q> = <<Q as WorldQuery>::Fetch as Fetch<'w, 's>>::Item;
+pub type QueryFetch<'w, Q> = <Q as WorldQueryGats<'w>>::Fetch;
+pub type QueryItem<'w, Q> = <<Q as WorldQueryGats<'w>>::Fetch as Fetch<'w>>::Item;
+pub type ROQueryFetch<'w, Q> = <Q as WorldQueryGats<'w>>::ReadOnlyFetch;
+pub type ROQueryItem<'w, Q> = <<Q as WorldQueryGats<'w>>::ReadOnlyFetch as Fetch<'w>>::Item;
+
+pub trait WorldQueryGats<'world> {
+    type Fetch: Fetch<'world, State = Self::_State>;
+    type ReadOnlyFetch: Fetch<'world, State = Self::_State> + ReadOnlyFetch<'world>;
+    type _State: FetchState;
+}
 
 /// Types that implement this trait are responsible for fetching query items from tables or
 /// archetypes.
 ///
-/// Every type that implements [`WorldQuery`] have their associated [`WorldQuery::Fetch`] and
+/// Every type that implements [`WorldQuery`] have their associated [`WorldQuery::Fetch`](WorldQueryGats::Fetch)  and
 /// [`WorldQuery::State`] types that are essential for fetching component data.
-pub trait Fetch<'world, 'state>: Sized {
+pub trait Fetch<'world>: Sized {
     type Item;
     type State: FetchState;
 
     /// Creates a new instance of this fetch.
     ///
     /// # Safety
     ///
-    /// `state` must have been initialized (via [`FetchState::init`]) using the same `world` passed
-    /// in to this function.
+    /// `state` must have been initialized (via [`FetchState::init`]) using the same `world` passed in
+    /// to this function.
```

Comment:
```suggestion
    /// `state` must have been initialized (via [`FetchState::init`]) using the same `world` passed
    /// in to this function.
```

Just to reduce noise in this PR.